### PR TITLE
Added extra info for each Special Action

### PR DESCRIPTION
**Pull request by jarboo**
image
image
image

### CODE REVIEW

File: src/ImportExport.ts [hunk 1]
```diff
@@ -176,6 +178,37 @@ export const importSynergism = async (input: string, reset = false) => {
     }
 }
 
+export const promocodesInfo = async (input:string) => {
```
Comment (KhafraDev) [inline src/ImportExport.ts:181]:
```suggestion
export const promocodesInfo = async (input: string) => {
```

File: src/ImportExport.ts [hunk 2]
```diff
@@ -176,6 +178,37 @@ export const importSynergism = async (input: string, reset = false) => {
...
+    const textElement = DOMCacheGetOrSet('promocodeinfo');
+    let textMessage: string = '\'' + input +'\': ';
```
Comment (KhafraDev) [inline src/ImportExport.ts:183]:
```suggestion
    let textMessage = `'${input}': ';
```

File: src/ImportExport.ts [hunk 3]
```diff
@@ -176,6 +178,37 @@ export const importSynergism = async (input: string, reset = false) => {
...
+            }
+            break;
+        default:
+            textMessage = '';
+    }
+
+    textElement.textContent = textMessage;
+}
+
+
```
Comment (KhafraDev) [inline src/ImportExport.ts:210]:
```suggestion
```

**Review by KhafraDev — CHANGES_REQUESTED** [review]

**Review by KhafraDev — APPROVED** [review]

File: src/ImportExport.ts [hunk 4]
```diff
@@ -176,6 +178,35 @@ export const importSynergism = async (input: string, reset = false) => {
     }
 }
 
+export const promocodesInfo = async (input: string) => {
+    const textElement = DOMCacheGetOrSet('promocodeinfo');
+    let textMessage = `'${input}': `;
+    let availableUses = 0;
+    switch (input) {
+        case 'daily':
+            textMessage += (player.dailyCodeUsed ? '0' : '1') + ' use left. Next: end of the day.';
```
Comment (KhafraDev) [inline src/ImportExport.ts:187]: should be "uses" for 0, "use" for 1

File: src/ImportExport.ts [hunk 5]
```diff
@@ -176,6 +178,35 @@ export const importSynergism = async (input: string, reset = false) => {
...
+            availableUses = addCodeAvailableUses();
+            textMessage += availableUses + ' use' + (availableUses > 1 ? 's' : '') + ' left.';
+            if (availableUses == 0) {
+                textMessage += ' Next: in ' + addCodeTimeToNextUse() + ' seconds.';
+            }
+            break;
+        case 'time':
+            availableUses = timeCodeAvailableUses();
+            textMessage += availableUses + ' use left.';
+            if (availableUses == 0) {
```
Comment (KhafraDev) [inline src/ImportExport.ts:199]:
```suggestion
            if (availableUses === 0) {
```

File: src/ImportExport.ts [hunk 6]
```diff
@@ -176,6 +178,35 @@ export const importSynergism = async (input: string, reset = false) => {
...
+        case 'add':
+            availableUses = addCodeAvailableUses();
+            textMessage += availableUses + ' use' + (availableUses > 1 ? 's' : '') + ' left.';
+            if (availableUses == 0) {
+                textMessage += ' Next: in ' + addCodeTimeToNextUse() + ' seconds.';
+            }
+            break;
+        case 'time':
+            availableUses = timeCodeAvailableUses();
+            textMessage += availableUses + ' use left.';
```
Comment (KhafraDev) [inline src/ImportExport.ts:198]: likewise should be "uses" for 0, "use" for 1

**Review by KhafraDev — CHANGES_REQUESTED** [review]

File: src/ImportExport.ts (hunk 4 above)
Comment (jarboo) [reply src/ImportExport.ts:187]: "uses" for 0, really? English is not my native language, but this seems weird. How is 0 plural?

Comment (KhafraDev) [reply src/ImportExport.ts:187]: It's weird but correct, https://english.stackexchange.com/q/38293 has a better answer than I could give

Comment (jarboo) [reply src/ImportExport.ts:187]: https://www.quora.com/Why-is-0-plural
Ok it seems to be confirmed. I'm really surprised I've never heard of this before, it feels super werid ^^
Do you want me to change the code again, then? It's already approved, no?

Comment (KhafraDev) [reply src/ImportExport.ts:187]: Yes please, there are some other comments that I missed below too (you might need to refresh the page)

**Pull request merged**